src/NLog/Targets/FileTarget.cs

```diff
@@ -1861,6 +1933,9 @@ private void DeleteOldArchiveFiles()
                     try
                     {
                         File.Delete(archiveFileName);
+                        var preventNullRefs = this.OnArchiveFileDeleted;
+                        if (preventNullRefs != null)
+                            preventNullRefs(archiveFileName);
```

Comment:
Code duplication. Should be call to `fileArchive_OnArchiveFileDeleted` / `fileArchive_OnFileArchived`?